pkg/controller/versions/strategy_v2_3.go

```diff
@@ -351,6 +354,35 @@ func (v *versionStrategyV2_3) Render(ctx context.Context, cr *common.ControllerR
 			}
 		}
 	}
+
+	if isComponentEnabled(spec.Istio, v2_3ChartMapping[PrometheusChart].enabledField) {
+		members := &v1.ServiceMeshMemberList{}
+
+		err := cr.Client.List(ctx, members, client.MatchingFields{"spec.controlPlaneRef.namespace": smcp.GetNamespace()})
```

Comment:
as discussed on the call, we should be `Get()`ing the `MemberRoll` here and look at its `.status.configuredMembers` instead of listing the `Members`